packages/hash/frontend/src/pages/[account-slug]/entities/[entity-id].page/entity-editor/links-section.tsx

```diff
@@ -19,13 +20,15 @@ different entities"
 );
 
 export const LinksSection = () => {
-  const { entity } = useEntityEditor();
+  const { entityRootedSubgraph } = useEntityEditor();
   const [showSearch, setShowSearch] = useState(false);
 
-  if (!entity) {
+  if (!entityRootedSubgraph) {
     return null;
   }
 
+  const entity = rootsAsEntities(entityRootedSubgraph)[0]!;
```

Comment:
My understanding from a prior conversation is that when an entity is not found, an empty subgraph would be returned. That means that the `entityRootedSubgraph` would be truthy, whereas `entity` would've previously been falsy.

If my understanding is correct I think this therefore introduces a run-time error where there previously wouldn't have been one, as when an entity is not found `null` wouldn't be returned - an index out of bounds error would be thrown when you try accessing `rootsAsEntities(entityRootedSubgraph)[0]`.

Can we therefore instead check that the entity roots array contains more than one element, and return `null` otherwise so that this doesn't introduce potential issues in existing code?

(this applies to various other files that have been modified, maybe you'll therefore want to encapsulate this in a `getSingleEntityRoot` method that returns `null` or the root entity, and throws if there's more than one entity in the roots array)

Reply:
You are correct, thank you for spotting! My stupid use of `!` would have made this really annoying to find too.

I'm a little wary of creating a tonne of permutations of `getSingleXRoot`, so I'm trying to think of another way we can do this. I've been floating with an idea of creating a 'view' class over the `subgraph` like `EntityRootedSubgraph` which when you construct it, it validates there are roots and they are all entities, and then you never need to check that again. And we could add normal methods like `.roots` which returns `Entity[]` without the need for type guards, etc.

Reply:
The simplest thing would be just to check the array length before accessing, I don't feel strongly about adding something re-usable here at this stage.

Reply:
Okay lots of iteration on this and I couldn't find a single thing I was happy with. I think the only solution that was clean and didn't involve a huge amount of permutations or generics or abstract classes was basically adding a property to the object. I have some questions about the ergonomics or viability of this but I will open separate threads for those.

https://github.com/hashintel/hash/pull/1242/commits/6cdac8f23e0635cabc1f1d75c91cc9a06d96e699